### PR TITLE
migrate the K8s networking model section

### LINKED ISSUE #25832: Migrate “The Kubernetes network model” into “Services, Load Balancing, and Networking” section
**This is a Feature Request**

**What would you like to be added**
Take the section https://kubernetes.io/docs/concepts/cluster-administration/networking/#the-kubernetes-network-model (just the heading and the text up to the next heading) and move that text into the page at https://kubernetes.io/docs/concepts/services-networking/

In place of that heading and text, add a link to its new location.

**Why is this needed**
[Services, Load Balancing, and Networking](https://kubernetes.io/docs/concepts/services-networking/) is too terse, and does not introduce readers into important networking concepts.

**Comments**
/language en
/sig network

### PR DESCRIPTION
**Pull request by neha-viswanathan**
Fixes #25832

### CODE REVIEW

**Review by sftim — COMMENTED** [review]
@shannonxtreme would you be willing to take a look at this pull request? I'd be keep to understand how that fits in with some changes you're considering in this section.

**Comment by sftim** [comment]
LGTM

**Comment by shannonxtreme** [comment]
> @shannonxtreme would you be willing to take a look at this pull request? I'd be keep to understand how that fits in with some changes you're considering in this section.

Not sure how I feel about having the TOC on that page when there is text on it. Sidebar nav should be enough (similar to https://kubernetes.io/docs/concepts/workloads/)

---

This can be a separate PR though:

I think it would benefit from a diagram as well to visualize the networking model. Similar to the Components/Cluster Architecture ones where the overview has a basic diagram that drills deeper into each component in the relevant section. 

[This medium post](https://medium.com/google-cloud/understanding-kubernetes-networking-pods-7117dd28727) has some nice diagrams that are broken into each component.

**Comment by sftim** [comment]
One thing I'll point out: the sidebar is visible to desktop clients and on big tablets. On a mobile or portrait-oriented device, you might not see it.

**Comment by sftim** [comment]
:+1: to suppressing the automatic TOC (use `no_list: true` in the front matter).

/lgtm

**Comment by kbhawkey** [comment]
👋 @neha-viswanathan . Thanks for your contribution.
Because the _index page contains a somewhat lengthy TOC (12 links), I don't know if it makes sense to move the model content to that page. Could this content move to a new page or another page? What do you think?

Page preview:
https://deploy-preview-28617--kubernetes-io-main-staging.netlify.app/docs/concepts/services-networking/#the-kubernetes-network-model

**Comment by neha-viswanathan** [comment]
> I don't know if it makes sense to move the model content to that page. Could this content move to a new page or another page? 

Hi @kbhawkey 🙋🏽‍♀️ I followed the suggestion from @sftim on using `no_list: true`  so the TOC doesn't appear on the page anymore.

**Comment by sftim** [comment]
Setting `no_list: true` makes sense _only_ when we provide equivalent links within the page text. This is because not all ways of viewing the docs show the left-hand sidebar.

**Comment by shannonxtreme** [comment]
Agreeing with Tim. Can we include organic links to topics in this section, similar to https://kubernetes.io/docs/concepts/containers/? LGTM to me after that

**Comment by kbhawkey** [comment]
Hi @neha-viswanathan . The updates look good.
The commits include a couple of extra directory changes (submodules).
Can you remove the `api-ref-generator` and `docsy` submodules?
Thanks!

**Comment by neha-viswanathan** [comment]
@shannonxtreme @kbhawkey ready for review again :)

File: content/en/docs/concepts/services-networking/_index.md [hunk 1]
```diff
@@ -3,8 +3,51 @@ title: "Services, Load Balancing, and Networking"
...
+How this is implemented is a detail of the particular container runtime in use.
+
+It is possible to request ports on the `Node` itself which forward to your `Pod`
+(called host ports), but this is a very niche operation. How that forwarding is
+implemented is also a detail of the container runtime. The `Pod` itself is
+blind to the existence or non-existence of host ports.
+
 Kubernetes networking addresses four concerns:
 - Containers within a Pod use networking to communicate via loopback.
 - Cluster networking provides communication between different Pods.
```
Comment (shannonxtreme) [inline content/en/docs/concepts/services-networking/_index.md:53]: Cant review it for some reason but:

 line 54, link `expose an application running in Pods` to https://kubernetes.io/docs/concepts/services-networking/connect-applications-service/

line 54, link `Service resource` to https://kubernetes.io/docs/concepts/services-networking/service/

Line 55, maybe link `publish services only for consumption inside your cluster` to https://kubernetes.io/docs/concepts/services-networking/service-traffic-policy/

Line 52: link `use networking to communicate` to https://kubernetes.io/docs/concepts/services-networking/dns-pod-service/

File: content/en/docs/concepts/services-networking/_index.md [hunk 2]
```diff
@@ -3,8 +3,51 @@ title: "Services, Load Balancing, and Networking"
...
 ---
 
+## The Kubernetes network model
+
+Every [`Pod`](/docs/concepts/workloads/pods/) gets its own IP address. 
+This means you do not need to explicitly create links between `Pods` and you 
+almost never need to deal with mapping container ports to host ports.  
+This creates a clean, backwards-compatible model where `Pods` can be treated 
+much like VMs or physical hosts from the perspectives of port allocation, 
+naming, service discovery, load balancing, application configuration,
```
Comment (shannonxtreme) [inline content/en/docs/concepts/services-networking/_index.md:16]: Link "load balancing" to https://kubernetes.io/docs/concepts/services-networking/ingress/#load-balancing

**Review by shannonxtreme — COMMENTED** [review]
Suggestions for linking.

You can also just rollback `no_list=true` and keep the original TOC (and avoid having to use organic links). My original suggestion to remove the TOC was aesthetic but like @sftim said, it might be more accessible to keep it. 

Up to you. Either way, I'll lgtm after you implement whichever :)

**Comment by shannonxtreme** [comment]
I'm okay with this version being merged. 

/lgtm

**Review by shannonxtreme — COMMENTED** [review]
Good changes. I like it in the current state. Thanks, Neha!

**Review by mehabhalodiya — COMMENTED** [review]
LGTM!

**Comment by thockin** [comment]
/hold

I disagree with this change.  This takes the guts out of the page and puts them somewhere else.  This text isn't even about Services - it's primarily about Pods.

**Comment by sftim** [comment]
@thockin, thanks for your views here.

My take on this when I reviewed it was that the text headed “The Kubernetes network model” belongs within the **Services, Load Balancing, and Networking** section. I still think that's the best home.

If that text were already in **Services, Load Balancing, and Networking** and we'd proposed a move into **Cluster Administration**, would you support that move? I suspect you might not.

Please consider how this might look in reverse. If https://deploy-preview-28617--kubernetes-io-main-staging.netlify.app/docs/concepts/services-networking/ were the live page and the proposal were to turn it into https://k8s.io/docs/concepts/services-networking/ (moving that text into the page at https://deploy-preview-28617--kubernetes-io-main-staging.netlify.app/docs/concepts/cluster-administration/), would you genuinely recommend approving that change?

**Comment by sftim** [comment]
Issue https://github.com/kubernetes/website/issues/30745 is slightly related to this change, but only slightly.

**Comment by shannonxtreme** [comment]
I agree with @sftim. 

I'd expect this to be a top-level concept instead of two levels down (Concepts > Services, LB, Networking instead of Concepts > Cluster admin > Networking). The content in that section *does* talk about Pod DNS (in addition to the other types of networking).

Aside: I also think we should find a way to get rid of *Cluster admin > Networking* after we merge this PR, since that page will solely be third-party tools. We could redirect that TOC entry to the page in this PR.

**Comment by thockin** [comment]
It still feels out of place to me, but OK.  Understanding and fulfilling this is still something people struggle with all the time.

/lgtm
/approve

**Comment by kbhawkey** [comment]
I'll remove the hold.
/hold cancel

**Comment by kbhawkey** [comment]
Hi @neha-viswanathan .
Can you look at the submodules again (remove changes from this PR)? I would expect to only see two files updated. Thanks.
/hold

**Comment by neha-viswanathan** [comment]
> Hi @neha-viswanathan . The updates look good.
The commits include a couple of extra directory changes (submodules).
Can you remove the api-ref-generator and docsy submodules?

@kbhawkey Hello, based on your previous comment, I removed the submodules.

**Comment by kbhawkey** [comment]
> > Hi @neha-viswanathan . The updates look good.
> > The commits include a couple of extra directory changes (submodules).
> > Can you remove the api-ref-generator and docsy submodules?
> 
> @kbhawkey Hello, based on your previous comment, I removed the submodules.

Hello @neha-viswanathan .
I don't think you want to include any changes to the api-ref-generator and docsy submodules in this pull request? If that is the case, you can remove any extraneous commits in this PR and rebase your changes to a single commit. Sounds good?
/hold cancel

**Comment by sftim** [comment]
Taking LGTM from https://github.com/kubernetes/website/pull/28617#issuecomment-999114427

/lgtm
/approve

Thanks folks!

**Pull request merged**